include/openmc/material.h

```diff
@@ -90,6 +90,9 @@ class Material {
   void set_densities(
     const vector<std::string>& name, const vector<double>& density);
 
+  //! Clone the material by deep-copying all members
```

Comment:
Could you document this a little more to specify what is actually happening in this function? Sorry to overcomplicate it, but now that it returns a reference, it's not immediately clear that this function has the side effect that this new material is appended to the simulation's material vector. Also, it would be good to specify the exception to the deep copy, namely that the ID will be set to C_NONE.

Reply:
Yes, good point, that doc was not particularly descriptive.